### ISSUE TITLE
Jpeg image size

### ISSUE THREAD

**Issue by kelilevi** [libvips/ruby-vips#247]
Trying to use the ruby-vips .jpeg saved image in Onyx RIP software, the image sizes are different. 
Ruby-vips saved “.tif” and any Photoshop image (.jpg, .tif) size is right: 10 x 10, the ruby-vips saved .jpg size is showing: 20.83 x 20.83 and is ripping that size.
Attached are the ruby-vips generated "10x10_Vips.jpg" and the "10x10_from_Photoshop.jpg" images: 

[10x10_Vips_and_Photoshop.zip](https://github.com/libvips/ruby-vips/files/5212146/10x10_Vips_and_Photoshop.zip)

When running the vipsheader -a I am getting:

![image](https://user-images.githubusercontent.com/62903484/93007983-6be1ed00-f53d-11ea-955d-dca869c9dcfb.png)

Tried to remove all the “data”, tried to make a copy including the x/y-res – same size issue.
```
#!/usr/bin/ruby
ENV["VIPS_WARNING"] = '1' # For ignoring VIPS-WARNING: Incompatible type for "RichTIFFIPTC" - for tif images
require 'vips'

im = Vips::Image.new_from_file "./10x10.tif"  # "./10x10.jpg"

#im.remove "xmp-data"
#im.remove "photoshop-data"
#im.remove 'jpeg-thumbnail-data'
#im.remove "exif-data"

#final_dpi = 150
#new_image = im.copy xres: final_dpi / 25.4, yres: final_dpi / 25.4 # 150 = 5.9055118560791, 300 = 11.811023622047
#new_image.jpegsave "./10x10_Vips.jpg", profile: './AdobeRGB1998.icc', Q: 95

im.jpegsave "./10x10_Vips.jpg", profile: './AdobeRGB1998.icc', Q: 95

#im.tiffsave "./10x10_Vips.tif", profile: './AdobeRGB1998.icc', xres: final_dpi / 25.4, yres: final_dpi / 25.4, resunit: 'inch'
```
Tried to "set" some of the missing tag (exif-ifd1-XResolution, exif-ifd0-ImageWidth), but I am getting the "unimplemented gtype for set" error. Is there any way to write a new tag in the exif data?
```
new_image = im.copy()
tag = 'exif-ifd1-XResolution'
new_image.set(tag, ' 72/1')
```
Tried
`C:\vips-dev-8.10\bin\vips.exe magicksave "D:/Images/10x10.tif" "D:/Images/10x10_vips_1.jpg"`
same size issue...

Any help or guide to the right direction is appreciated. Thank you.

**Comment by jcupitt** [comment]
I think I've found it. JPG files have two different ways of attaching resolution information: as a JFIF tag, and as part of EXIF. libvips was setting both, but leaving the JFIF tags at their default values of 72 DPI. Most loaders will prefer the EXIF resolution, but your RIP seems to pick JFIF first.

I've changed libvips to not output the JFIF tags if it sets EXIF resolution. This change should be in 8.10.2.

Thanks for the report!
